www/searchutil.php

```diff
@@ -609,6 +610,16 @@ function doSearch($db, $term, $searchType, $sortby, $limit, $browse)
                                   . "and gpl.userid = '$txt'";
                 }
                 break;
+            
+            case 'competitionid':
+                // need to join the gameprofilelinks table to do this query
+                if (!isset($extraJoins[$col])) {
+                    $extraJoins[$col] = true;
+                    $tableList .= " inner join compgames "
+                                  . "on compgames.gameid = games.id "
+                                  . "and compgames.compid = '$txt'";
```

Comment:
Does `$txt` need to be escaped with `$txt = mysql_real_escape_string($txt, $db);`?

It's done in several other places, but not in `authorid` and here.

Is it guaranteed to be safe as part of an earlier parsing step? I can't find it.

Reply:
Nope! It's just 100% vulnerable. 😬